river/Cursor.zig

```diff
@@ -133,6 +133,19 @@ pub fn deinit(self: *Self) void {
     self.wlr_cursor.destroy();
 }
 
+/// Aborts all operations and updates the surface beneath the cursor. This is
+/// intended to be used when the output or tag focus changes.
+pub fn resetState(self: *Self) void {
+    log.debug("reset state, current mode is {s}", .{@tagName(self.mode)});
+
+    // Go back to .passthrough mode. Note that if the previous mode was .down,
+    // we do not send button release events. This is on purpose.
+    self.mode = .passthrough;
+
+    const now = util.getNowMsec() catch return;
+    self.passthrough(now);
```

Comment:
I wonder if we can instead get the time from the event that caused the tag/output focus change. Might be annoying to pass around though.

Reply:
I don't think that would really be worth it.